### PR TITLE
[libcxx][test][AIX] address more platform differences in locale tests

### PR DESCRIPTION
**Pull request by daltenty**
This is a follow on to https://github.com/llvm/llvm-project/pull/92312, where we address some more locale  platform differences. These are:

- for locale fr_FR AIX libc expects `U202F` as `LC_MONETARY` `thousands_sep`
- for locale zh_CN AIX libc `LC_MONETARY` has `n_sign_posn == 1`, indicating the `negative_sign` should come before the `currency_symbol` string

### CODE REVIEW

File: libcxx/test/std/localization/locale.categories/category.monetary/locale.money.get/locale.money.get.members/get_long_double_zh_CN.pass.cpp [hunk 1]
```diff
@@ -204,17 +209,20 @@ int main(int, char**)
...
-            long double ex;
-            std::ios_base::iostate err = std::ios_base::goodbit;
-            I iter = f.get(I(v.data()), I(v.data() + v.size()),
-                                                false, ios, err, ex);
-            assert(base(iter) == v.data() + v.size());
-            assert(err == std::ios_base::eofbit);
-            assert(ex == -123456789);
-            std::noshowbase(ios);
+#ifdef _AIX
+          std::string v = "-" + currency_symbol + "1,234,567.89";
```
Comment (ldionne) [inline libcxx/test/std/localization/locale.categories/category.monetary/locale.money.get/locale.money.get.members/get_long_double_zh_CN.pass.cpp:213]: I would argue this is a bug in the C library on AIX, is that not possible? If so, we should instead fix this on AIX and mark the test as XFAIL on the versions of AIX where the fix hasn't made it yet.

**Review by ldionne — CHANGES_REQUESTED** [review]

File: libcxx/test/std/localization/locale.categories/category.monetary/locale.money.get/locale.money.get.members/get_long_double_zh_CN.pass.cpp (hunk 1 above)
Comment (ldionne) [reply libcxx/test/std/localization/locale.categories/category.monetary/locale.money.get/locale.money.get.members/get_long_double_zh_CN.pass.cpp:213]: Ack. I'm concerned about the maintainability of our current approach, but you're right that this patch doesn't make things significantly worse nor does it stray from the status quo.

**Review by ldionne — APPROVED** [review]

File: libcxx/test/std/localization/locale.categories/category.monetary/locale.money.get/locale.money.get.members/get_long_double_zh_CN.pass.cpp (hunk 1 above)
Comment (mordante) [reply libcxx/test/std/localization/locale.categories/category.monetary/locale.money.get/locale.money.get.members/get_long_double_zh_CN.pass.cpp:213]: In my experience locale definitions are not as "universal" as expected. So we have several tests that hard-code what is used on a platform. I'm a bit surprised `U202F` was not used already. For example, `libcxx/test/std/localization/locale.categories/facet.numpunct/locale.numpunct.byname/thousands_sep.pass.cpp` already does.

**Review by xingxue-ibm — APPROVED** [review]
LGTM, thanks!

File: libcxx/test/std/localization/locale.categories/category.monetary/locale.money.get/locale.money.get.members/get_long_double_zh_CN.pass.cpp (hunk 1 above)
Comment (daltenty) [reply libcxx/test/std/localization/locale.categories/category.monetary/locale.money.get/locale.money.get.members/get_long_double_zh_CN.pass.cpp:213]: > I'm a bit surprised U202F was not used already. For example, libcxx/test/std/localization/locale.categories/facet.numpunct/locale.numpunct.byname/thousands_sep.pass.cpp already does.

Yeah, it seems to be the case that some tests put that platform specific knowledge into a common place and others `platform_support.h` and others just hardcode it in the individual test, so even though we fixed in one test others are still affected. I think cleaning that up could help the maintainability concerns here. I'll try to take a look into that as a follow on to this PR.

**Pull request merged**